CMakeLists.txt

```diff
@@ -53,7 +53,7 @@ else()
     find_package (Threads)
 
     set( CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -Wpedantic -Wno-unused-parameter")
+      "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -Wpedantic -Wno-unused-parameter -Wno-implicit-fallthrough")
```

Comment:
I prefer not to turn off the warning, I believe the remaining warnings are fixed in 154.

Reply:
The `CMAKE_CXX_FLAGS` should be set by the user(either from the command-line or in the toolchain file).  You can add flags for warnings with `add_compile_options`.

Also, you shouldn't set the C++ version. The user should decide that. You could set a default with `CMAKE_CXX_STANDARD`.

Reply:
But this CMakeLists.txt is for building the tests and examples, not for use for building a user's project that uses beast.

Reply:
But you want to be able build the tests and examples with a specific C++ version. There is semantic differences between the C++11 and C++17 that it makes sense to want to build the  tests and examples with C++17 instead of C++11.

Reply:
I want to make sure that the tests and examples compile in C++11, so I don't accidentally use a C++14 or C++17 feature when I am maintaining beast.

Reply:
Yea but you have no way of testing for C++14 or C++17, either. Should users assume these are not supported since they are never tested for? 

Testing with C++11 will ensure everything will pass with C++11, but I dont see why that would negate testing with newer version as well.